cleanlab_studio/studio/studio.py

```diff
@@ -127,34 +128,29 @@ def apply_corrections(self, cleanset_id: str, dataset: Any, keep_excluded: bool
         label_column = api.get_label_column_of_project(self._api_key, project_id)
         id_col = api.get_id_column(self._api_key, cleanset_id)
         if _pyspark_exists and isinstance(dataset, pyspark.sql.DataFrame):
-            from pyspark.sql.functions import udf
-
-            cl_cols = self.download_cleanlab_columns(cleanset_id, to_spark=True)
+            cl_cols = self.download_cleanlab_columns(cleanset_id, include_project_details=True, to_spark=True)
             corrected_ds_spark = dataset.alias("corrected_ds")
             if id_col not in corrected_ds_spark.columns:
                 from pyspark.sql.functions import (
                     row_number,
                     monotonically_increasing_id,
+                    when,
+                    col
                 )
                 from pyspark.sql.window import Window
 
                 corrected_ds_spark = corrected_ds_spark.withColumn(
                     id_col,
                     row_number().over(Window.orderBy(monotonically_increasing_id())) - 1,
                 )
-            both = cl_cols.select([id_col, "action", "clean_label"]).join(
+            both = cl_cols.select([id_col, "action", "corrected_label"]).join(
                 corrected_ds_spark.select([id_col, label_column]),
                 on=id_col,
                 how="left",
             )
             final = both.withColumn(
                 "__cleanlab_final_label",
-                # XXX hacky, checks if label is none by hand
-                # instead, use original JSON, which uses null values where it's not specified
-                udf(lambda original, clean: original if check_none(clean) else clean)(
-                    both[label_column],
-                    "clean_label",
-                ),
+                when(col("clean_label").isNull(), col(label_column)).otherwise(col("clean_label"))
```

Comment:
Thanks for figuring this out! Much better than the original `udf` implementation. Only issue here is I believe should be using "corrected_label" instead of "clean_label".